pyqtgraph/functions.py

```diff
@@ -1269,6 +1274,50 @@ def makeARGB(data, lut=None, levels=None, scale=None, useRGBA=False, output=None
     return imgData, alpha
 
 
+def try_fastpath_argb(xp, ain, aout, useRGBA):
+    # we only optimize for certain cases
+    # return False if we did not handle it
+    can_handle = xp is np and ain.dtype == xp.ubyte and ain.flags['C_CONTIGUOUS']
+    if not can_handle:
+        return False
+
+    nrows, ncols = ain.shape[:2]
+    nchans = 1 if ain.ndim == 2 else ain.shape[2]
+
+    Format = QtGui.QImage.Format
+
+    if nchans == 1:
+        in_fmt = Format.Format_Grayscale8
+    elif nchans == 3:
+        in_fmt = Format.Format_RGB888
+    else:
+        in_fmt = Format.Format_RGBA8888
+
+    if useRGBA:
+        out_fmt = Format.Format_RGBA8888
+    else:
+        out_fmt = Format.Format_ARGB32
+
+    if in_fmt == out_fmt:
+        aout[:] = ain
+        return True
+
+    npixels_chunk = 512*1024
+    batch = int(npixels_chunk / ncols / nchans)
+    batch = max(1, batch)
+    row_beg = 0
+    while row_beg < nrows:
```

Comment:
can't help but look at this and think of your implementation of `np.nditer` simplifying this a fair amount.  Is there a use-case here for that?

Reply:
nditer is meant for cases where each element in the uint8 array can be processed independently. This is true for nchans==1 and could perhaps be done for nchans==4 (RGBA) by viewing the array as uint32. I don't think it can be done for the case nchans==3 (RGB).